llvm/test/CodeGen/Mips/fp-maximumnum-minimumnum.ll

```diff
@@ -96,17 +285,89 @@ define float @minimumnum_float_nnan(float %x, float %y) {
 ; MIPS32R6:       # %bb.0:
 ; MIPS32R6-NEXT:    jr $ra
 ; MIPS32R6-NEXT:    min.s $f0, $f12, $f14
+;
+; MIPS64-LABEL: minimumnum_float_nnan:
+; MIPS64:       # %bb.0:
+; MIPS64-NEXT:    c.olt.s $f12, $f13
+; MIPS64-NEXT:    mov.s $f0, $f13
+; MIPS64-NEXT:    movt.s $f0, $f12, $fcc0
+; MIPS64-NEXT:    mfc1 $1, $f12
+; MIPS64-NEXT:    lui $2, 32768
+; MIPS64-NEXT:    xor $1, $1, $2
+; MIPS64-NEXT:    mov.s $f1, $f0
+; MIPS64-NEXT:    movz.s $f1, $f12, $1
+; MIPS64-NEXT:    mfc1 $1, $f13
+; MIPS64-NEXT:    xor $1, $1, $2
+; MIPS64-NEXT:    movz.s $f1, $f13, $1
+; MIPS64-NEXT:    mtc1 $zero, $f2
+; MIPS64-NEXT:    c.eq.s $f0, $f2
+; MIPS64-NEXT:    jr $ra
+; MIPS64-NEXT:    movt.s $f0, $f1, $fcc0
+  %z = call nnan float @llvm.minimumnum.f32(float %x, float %y)
```

Comment:
I'd prefer to test these with some simpler to understand operation than minimumnum. Is there some other fold that doesn't involve any flavor of fmin?

Reply:
I can only find there is only one about FP_TO_BF16.

Reply:
The fcmp + select to min variant would be more straightforward

Reply:
I have a try `fcmp + select` with this IR
```
define dso_local noundef float @max(float noundef nofpclass(nan) %a, float noundef nofpclass(nan) %b) local_unnamed_addr #0 {
entry:
  %cmp = fcmp ogt float %a, %b
  %cond = select i1 %cmp, float %a, float %b
  ret float %cond
}
```

It is not easy to support it. The problem is that the argument of `void SelectionDAGBuilder::visitSelect(const User &I)`
is marked as `const`, so we have no chance to edit `I` if we detect the both `LHS` and `RHS` are `nnan`.

And then `visitSelect` calls
```
SelectPatternResult llvm::matchDecomposedSelectPattern(
    CmpInst *CmpI, Value *TrueVal, Value *FalseVal, Value *&LHS, Value *&RHS,
    Instruction::CastOps *CastOp, unsigned Depth)
```
It may be not a good idea to use the methods from SelectionDAG there.

Reply:
@arsenm https://github.com/llvm/llvm-project/pull/109570  To use `fcmp+select`, we need this PR first.

Reply:
You can use other incoming instructions with nnan flags, this PR is not necessary for test coverage

Reply:
> You can use other incoming instructions with nnan flags, this PR is not necessary for test coverage

ISD::SELECT seems ignoring `nnan`.

Reply:
Means there's a flag dropped or SelectionDAG never learned about select flags. Try something like an fadd?